src/core/blueprint/operators/apollo_federation.rs

```diff
@@ -36,8 +36,9 @@ pub fn compile_entity_resolver(inputs: CompileEntityResolver<'_>) -> Valid<IR, S
                 Resolver::Http(http) => compile_http(
                     config_module,
                     http,
+                    &Default::default(),
                     // inner resolver should resolve only single instance of type, not a list
-                    false,
```

Comment:
We need to guarantee over here that the entity isn't a list. Why are we overriding it inside this the function?

Reply:
I think there is a confusion.

Earlier the boolean was passed to check if the field `is_list`

but now I am passing the entire field and checking `field.type_of.is_list()`. In the case of federation, the boolean is replaced by `Field::default()` so `field.type_of.is_list()` will be false by default.

Now, I added the 4th parameter i.e. `is_federation` which is to skip the checks on args, as there are no args when http directive is defined on a Type.